### DISCUSSION TITLE
how do I reduce memory usage?

### DISCUSSION THREAD

**Discussion by rtwfroody** [gatsbyjs/gatsby#28677] [Help]
I'm running gatsby on a small VM I rent. It typically has a little over 3GB of RAM free. Ever since the recent new release, gatsby has a tendency to be killed by Linux' out-of-memory killer. What can I do to make gatsby use less RAM?

**Comment by vladar** [comment]
How many pages do you have? How many images do you have? Can you provide access to your website repository? CC @pvdz

**Reply by rtwfroody** [reply]
Is there a way to have gatsby tell me the answer to those questions? I think there are about 300 pages and about 1000 images.

Here's `ls -lR` for the directory (excluding public): 
[ls.txt](https://github.com/gatsbyjs/gatsby/files/5717158/ls.txt)

I don't have a git repo with the contents or anything, but I made you a tar ball here: http://casualhacker.net/tmp2/casualhacker.tgz

This gave me the following error just now:
[crash.txt](https://github.com/gatsbyjs/gatsby/files/5717241/crash.txt)

**Reply by rtwfroody** [reply]
It looks like the problem is that I have plotly-latest.min.js under src/pages. I intended to use it as a library but gatsby is interpreting it as a page, and whatever that involves chews through all the memory. I've moved it away and things seem OK.

**Comment by rtwfroody** [comment]
This VM has 2 cores, but setting GATSBY_CPU_COUNT=1 doesn't help.
